ppa/archive/gale.py

```diff
@@ -218,19 +223,34 @@ def get_item_pages(self, item_id, gale_record=None):
         if gale_record is None:
             gale_record = self.get_item(item_id)
 
+        local_ocr_text = None
+        try:
+            # Use higher quality local OCR text if available
+            local_ocr_text = get_local_ocr(item_id)
+        except FileNotFoundError:
+            logger.warning(f"Local OCR not found for {item_id}")
+
         # iterate through the pages in the response
         for page in gale_record["pageResponse"]["pages"]:
             page_number = page["pageNumber"]
 
-            # Fetch higher quality local OCR text if possible, with fallback to Gale
-            # OCR. Set a tag to indicate the usage of local OCR text.
+            # Use local OCR text if we have it, with fallback to Gale
+            # OCR. Set a tag to indicate when local OCR is present.
             tags = []
-            try: 
-                ocr_text = get_local_ocr(item_id, page_number)
-                tags = ["local_ocr"]
-            except FileNotFoundError as e:
-                ocr_text = page.get("ocrText")  # some pages have no text
-                logger.warning(f'Local OCR not found for {item_id} {page_number}')
+            ocr_text = None
+            if local_ocr_text:
+                ocr_text = local_ocr_text.get(page_number)
+                # if we have content, set tag to indicate local ocr
+                if ocr_text:
+                    tags = ["local_ocr"]
```

Comment:
Should this be set if the resulting OCR for the page is blank?

Reply:
nope, good catch

Reply:
wait, maybe I misunderstood your question - that's what the logic is now, do you disagree with it?

Reply:
It seems like the tag should be set if the text is blank, because the empty string is a judgment of the local OCR (and not the Gale OCR)